nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/DocumentBrowsingTest.java

```diff
@@ -432,6 +433,64 @@ public void itCanSetSimplePropertyToNullAndAnotherOneToNonNullInsideComplexOne()
         }
     }
 
+    /*
+     * NXP-28298
+     */
+    @Test
+    public void itCanSetArrayPropertyToEmpty() throws IOException {
+        DocumentModel doc = session.createDocumentModel("/", "myDocument", "File");
+        doc.setPropertyValue("dc:subjects", new String[] { "foo" });
+        doc = session.createDocument(doc);
+        fetchInvalidations();
+
+        try (CloseableClientResponse response = getResponse(RequestType.PUT, "id/" + doc.getId(),
+                "{\"entity-type\":\"document\",\"properties\":{\"dc:subjects\":[]}}")) {
+
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonNode jsonProperties = node.get("properties");
+            assertNotNull(jsonProperties);
+            JsonNode jsonSubjects = jsonProperties.get("dc:subjects");
+            assertNotNull(jsonSubjects);
+            assertTrue(jsonSubjects.isArray());
+            assertEquals(0, jsonSubjects.size());
```

Comment:
```suggestion
            assertTrue(jsonSubjects.isEmpty());
```

Reply:
We can't use it as it is a 2.10 method in jackson while we depend on 2.9.